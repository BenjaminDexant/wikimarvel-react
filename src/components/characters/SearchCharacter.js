import React, { useState } from "react";
import { useForm } from "react-hook-form";

import DisplayCharacters from "./DisplayCharacters";

//import "./searchCharacter.css";

const SearchCharacter = () => {
	const [character, setCharacter] = useState("");
	const {
		register,
		handleSubmit,
		formState: { errors },
	} = useForm();
	const onSubmit = ({ characterName }) => setCharacter(characterName);

	return (
		<div>
			<form onSubmit={handleSubmit(onSubmit)}>
				<input {...register("characterName", { required: true })} />
				{errors.characterName && <p>Field is required.</p>}
				<input type="submit" />
			</form>
			{character ? <DisplayCharacters name={character} /> : null}
		</div>
	);
};

export default SearchCharacter;
