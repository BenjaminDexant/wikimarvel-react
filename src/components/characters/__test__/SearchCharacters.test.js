import React from "react";
import SearchCharacter from "../SearchCharacter";
import { fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";

test("The form is displayed correctly", () => {
	const component = renderer.create(<SearchCharacter />).toJSON();
	expect(component).toMatchSnapshot();
});

test("Change value is working correctly", () => {
	const { getByTestId } = render(<SearchCharacter />);
	const inputElem = getByTestId("characterName");

	expect(inputElem.value).toBe("");

	fireEvent.change(inputElem, {
		target: {
			value: "test",
		},
	});

	expect(inputElem.value).toBe("test");
});
