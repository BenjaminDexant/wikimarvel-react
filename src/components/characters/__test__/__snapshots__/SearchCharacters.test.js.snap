// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The form is displayed correctly 1`] = `
<div
  className="container"
>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="characterName"
    >
      Character name :
    </label>
    <input
      data-testid="characterName"
      id="characterName"
      name="characterName"
      onBlur={[Function]}
      onChange={[Function]}
      type="characterName"
    />
    <button
      data-testid="submitButton"
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
